### PR TITLE
fix: Get lib.deno_core.d.ts to parse correctly

### PR DESCRIPTION
**Pull request by espindola**
I have tried to used this to typecheck a project and TSC found issues
in lib.deno_core.d.ts itself.

### CODE REVIEW

File: core/lib.deno_core.d.ts [hunk 1]
```diff
@@ -126,7 +126,7 @@ declare namespace Deno {
 
     export type PromiseRejectCallback = (
       type: number,
-      promise: Promise,
+      promise: Promise<any>,
```
Comment (kitsonk) [inline core/lib.deno_core.d.ts:129]:
```suggestion
      promise: Promise<unknown>,
```

**Review by kitsonk — COMMENTED** [review]
Thanks for this.  One minor suggestion.

**Comment by bartlomieju** [comment]
@kitsonk is there a way we could test this file in CI to avoid such mistakes in the future?

File: core/lib.deno_core.d.ts (hunk 1 above)
Comment (espindola) [reply core/lib.deno_core.d.ts]: Done

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks

**Pull request merged**